apps/web/app/components/VideoPlayer/VideoPlayer.tsx

```diff
@@ -1,126 +1,180 @@
+
+      if (!script) {
+        script = document.createElement("script");
+        script.id = scriptId;
+        script.src = "//assets.mediadelivery.net/playerjs/playerjs-latest.min.js";
+        script.async = true;
+
+        const scriptLoadHandler = () => {
+          script.setAttribute("data-loaded", "true");
+          initializePlayer();
+        };
+
+        script.onload = scriptLoadHandler;
+        script.onerror = () => {
+          console.error("Failed to load PlayerJS script");
+          setIsLoading(false);
+        };
+
+        document.body.appendChild(script);
+      } else if (script.getAttribute("data-loaded") === "true") {
+        initializePlayer();
+      }
     }
-  }, [isAdmin, onVideoEnded]);
 
-  useEffect(() => {
-    const video = videoRef.current;
-    if (!video) return;
-
-    const handlePlay = () => setIsPlaying(true);
-    const handlePause = () => setIsPlaying(false);
-    const handleEnded = () => {
-      handleVideoEnd();
-    };
+    return () => {
+      isMountedRef.current = false;
 
-    video.addEventListener("play", handlePlay);
-    video.addEventListener("pause", handlePause);
-    video.addEventListener("ended", handleEnded);
+      if (iframe) {
+        iframe.removeEventListener("load", () => setIsLoading(false));
+      }
 
-    return () => {
-      video.removeEventListener("play", handlePlay);
-      video.removeEventListener("pause", handlePause);
-      video.removeEventListener("ended", handleEnded);
+      cleanupRef.current();
+
+      if (playerRef.current) {
+        try {
+          if (iframe && document.body.contains(iframe)) {
+            playerRef.current.destroy?.();
+          }
+        } catch (e) {
+          console.error("Error destroying player:", e);
+        } finally {
+          playerRef.current = null;
+        }
+      }
     };
-  }, [handleVideoEnd]);
+  }, [initialUrl, handleVideoEnded]);
 
   return (
-    <div
-      className="group relative w-full"
-      role="button"
-      tabIndex={0}
-      onClick={togglePlay}
-      onKeyDown={(event) => {
-        if (event.key === " ") {
-          togglePlay();
-        }
-      }}
-    >
-      <video ref={videoRef} width="100%" height="auto">
-        <source src={url} />
-        <track kind="captions" />
-      </video>
-
-      <div
-        className={cn("absolute inset-0 bg-black transition-opacity duration-200", {
-          "opacity-48": !isPlaying,
-          "opacity-0 group-hover:opacity-48": isPlaying,
-        })}
+    <div className="relative aspect-video w-full">
+      {isLoading && (
+        <div className="absolute inset-0 flex items-center justify-center bg-gray-200">
+          <div className="h-12 w-12 animate-spin rounded-full border-b-2 border-t-2 border-blue-500"></div>
+        </div>
+      )}
+
+      <iframe
+        ref={iframeRef}
+        src={getPlayerUrl(initialUrl)}
+        className={`h-full w-full border-none ${isLoading ? "opacity-0" : "opacity-100"}`}
```

Comment:
Please use `cn` function instead of template literals
```suggestion
        className={cn("size-full border-none" {
          "opacity-0": isLoading,
          "opacity-100": !isLoading
        })}
```